### ISSUE TITLE
You must input `denatured_alcohol` at first!

### ISSUE THREAD

**Issue by NappingOcean** [cataclysmbnteam/Cataclysm-BN#4860]
### Describe the bug

I can't reload alcohol-fueled soldering iron with methanol until denatured alcohol has been reloaded and removed.
I suspect that [character_functions.cpp](https://github.com/cataclysmbnteam/Cataclysm-BN/blob/69f4a6a9f3a442629f756f1002498292af624b3f/src/character_functions.cpp) is in trouble.

### Steps To Reproduce

1. Make a player.
2. Spawn `alcohol-fueled soldering iron` and `methanol`.
3. Try to reload the iron with `methanol`.
4. Then spawn `denatured alcohol` and reload with it.
5. Remove `denatured alcohol` and repeat 3.

### Screenshots

![image](https://github.com/cataclysmbnteam/Cataclysm-BN/assets/129575271/b7a83fc0-7a57-4632-80db-569b28170b08)

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.19045.4529 (22H2)
- Game Version: f1e2f47 [64-bit]
- Graphics Version: Tiles
- LAPI Version: 2
- Game Language: English [en_US]
- Mods loaded: [
    Bright Nights [bn],
    Disable NPC Needs [no_npc_food],
    Simplified Nutrition [novitamins],
    No Rail Stations [No_Rail_Stations],
    Prevent Zombie Revivication [no_reviving_zombies],
    Limit Fungal Growth [limit_fungal_growth],
    Prevent Zombification [no_zombify],
    Stats Through Kills [stats_through_kills],
    Stats Through Skills [StatsThroughSkills],
    Magical Nights [MagicalNights],
    Wield Wild Weapons [WIELD_WILD_WEAPONS]
]

**Comment by SilearFlare** [comment]
`Needs verification `

I can verify that you can't reload an alcohol fueled soldering iron with methanol.

Steps to reproduce: 

Craft the damn thing (do not spawn the damn thing, it spawns with it through the debug menu)

Then try to reload it with methanol. 
Doesn't work.

Here, pictograms for reference.

Crafted the thing, tried to reload with methanol, does not work.
![Screenshot from 2024-09-27 17-25-40](https://github.com/user-attachments/assets/ee5c3f66-c119-4699-9596-f88bce72f792)
Debugged myself some denatured alcohol, now it accepts both options.
![Screenshot from 2024-09-27 17-19-22](https://github.com/user-attachments/assets/401e50c3-334b-49c1-a962-ee3e63259752)

tldr: something with how the reload code is handled had a fucky wucky and it will only accept methanol if it is fed even an ounce of denatured alcohol first, despite both of them meeting the concentrated alcohol flag requirement, you can only reproduce this if you do not spawn the thing. Shoutout to the only other dude playing innawoods because this literally happens the second you try to use a soldering iron that isn't battery powered, and methanol is extracted from wood instaed of concentrated alcohol, so it's the only option in the wild short of making some boozeberry mixed alcohol and then concentrating that into denatured alcohol just to get the damn thing to work (alas, extremely impractical)
![Screenshot from 2024-09-27 17-11-31](https://github.com/user-attachments/assets/628eefb2-cb66-47b8-a31d-3a193ec17ed9)

pls fix cooldude thx

**Comment by RoyalFox2140** [comment]
> pls fix cooldude thx

So This needs to be pointed out. Summoning me to an issue report isn't going to get me to fix them. You may find it surprising but not all collaborators on the project can code well (or at all)

**Comment by RoyalFox2140** [comment]
As for people who actually CAN work on magazine code you're looking at asking BroadFlatNails on the Discord AKA Coolthulhu on the Github, Jove if this was related to item identity and a bug caused by it, or other devs if they're interested/willing/qualified. You can ask in Development and see who might know what's going on. A number of dev roles are also busy with their lives or other code work on the project.

**Comment by RoyalFox2140** [comment]
![image](https://github.com/user-attachments/assets/7ab1f3ba-489e-4dc0-8bdb-6d06bdb55262)

Looks like the ammo type is copy-from'd. I wonder if this is related.
